### PR TITLE
Add CIDv0 RPC

### PR DESCRIPTION
**Pull request by keorn**

### CODE REVIEW

File: rpc/src/v1/impls/light/parity.rs [hunk 1]
```diff
@@ -342,4 +342,8 @@ impl Parity for ParityClient {
 			capability: Capability::Light,
 		})
 	}
+
+	fn ipfs_cid(&self, _content: Bytes) -> Result<String, Error> {
+		Err(errors::light_unimplemented(None))
```
Comment (rphmeier) [inline rpc/src/v1/impls/light/parity.rs]: why not implement this for the light client as well?

File: rpc/src/v1/helpers/errors.rs [hunk 2]
```diff
@@ -224,6 +225,14 @@ pub fn encryption_error<T: fmt::Debug>(error: T) -> Error {
 	}
 }
 
+pub fn encoding_error<T: fmt::Debug>(error: T) -> Error {
```
Comment (tomusdrw) [inline rpc/src/v1/helpers/errors.rs]: `_error` suffix is not really necessary

Comment (keorn) [inline rpc/src/v1/helpers/errors.rs]: Should I change all the other functions in this module too?

Comment (gavofyork) [inline rpc/src/v1/helpers/errors.rs]: if so, worth a separate PR.

**Comment by gavofyork** [comment]
does "add it for the light client as well" really mean a verbatim duplicate of that function? suffice it to say i consider this rather bad practice.

**Comment by rphmeier** [comment]
It's mostly a symptom of the fact that the `parity_` API does just about everything in a single trait. Doing an `Eth`, `EthFilter` split into `ParityChain` and `ParityMisc` would let us at least make `ParityMisc` generic.

**Comment by keorn** [comment]
@rphmeier Dont try to defend, that was just embarrassing...

**Pull request merged**